packages/account/src/providers/global-cache.ts

```diff
@@ -0,0 +1,41 @@
+import type { CachedResource } from './resource-cache';
+
+// Singleton class to manage global cache state
+export class GlobalCache {
+  private static instance: GlobalCache;
+  private cache: Map<string, CachedResource>;
+  private ttl: number;
+
+  private constructor() {
+    this.cache = new Map();
+    this.ttl = 0; // Will be set by the first Provider instance
+  }
+
+  static getInstance(): GlobalCache {
+    if (!GlobalCache.instance) {
+      GlobalCache.instance = new GlobalCache();
+    }
+    return GlobalCache.instance;
+  }
+
+  reset() {
+    GlobalCache.instance = new GlobalCache();
+    this.cache = new Map();
+    this.ttl = 0;
+  }
+
+  setTTL(ttl: number) {
+    // Only set TTL if not already set or if new TTL is smaller
+    if (this.ttl === 0 || ttl < this.ttl) {
+      this.ttl = ttl;
+    }
```

Comment:
Why is the TTL only updated if the provided value is lower than the previously set one?

Reply:
This is a preventative measure by taking a just conservative approach for a shared cache service. By keeping the smallest TTL it respects stricter freshness requirements. 

Here's an example scenario with a global caching strategy:

- User instantiates Provider 1 and Provider 2 with TTL = 60
- User updates Provider 1 to use TTL = 300

If we allow the larger TTL to override the smaller one, then Provider 2 might may have stale data (up to 300 seconds old) which would violate Provider 2's original requirement for 60 second freshness. In this case Provider 2 may reject a 299 second old UTXO for instance, when that wasn't necessarily desired when a user set a 60 second TTL. 

I think by taking this approach we give the user less chance of erring.

Reply:
@maschad This reasoning sounds counterintuitive and exposes the [negative] nuances of going down this opinionated caching rabbit hole. The interplay between instance and singleton cannot happen like this; it is confusing and would require an in-depth DX reevaluation.

@Torres-ssf @nedsalk I know you guys discussed this in the past, and although I can't remember exactly the context of where it came from, I must repeat my [YAGNI-inspired] statement from the last sync: I don't see a strong use case to justify such a feature right now.